DataStructures/disjoint.h

```diff
@@ -0,0 +1,20 @@
+#ifndef DISJOINT_H
+#define DISJOINT_H
+
+#include <cstdlib>
+#include <vector>
+
+class disjoint_set_t
```

Comment:
Why `disjoint_set_t`? Why not just `disjoint_set`, or better yet, `DisjointSet`. ([Type Names - Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Type_Names))